src/core.rs

```diff
@@ -8,27 +8,19 @@
 
 use byteorder::{ByteOrder, LittleEndian};
 
-/// Error that occured while performing and I2C Operation
-#[derive(Debug)]
-pub enum I2CError {
-    NotSupported,
-    Other(&'static str),
-}
-
-/// Result of an I2C Operation
-pub type I2CResult<T> = Result<T, I2CError>;
-
 /// Interface to an I2C Slave Device from an I2C Master
 ///
 /// Typical implementations will store state with references to the bus
 /// in use and the address of the slave device.  The trait is based on the
 /// Linux i2cdev interface.
 pub trait I2CDevice {
+    type Error;
```

Comment:
The other thing that could have an impact is if we want to support `no_std` so devices could be used, for instance, in something like zinc.  In that case, traits out of std (such as `Debug`) pose a problem.